pkg/api/job_runs.go

```diff
@@ -211,6 +211,17 @@ func JobRunRiskAnalysis(dbc *db.DB, jobRun *models.ProwJobRun, jobRunTestCount i
 				}
 			}
 
+			// otherwise, what is our best match...
+			// do something more expensive and check to see
+			// which tr contains all the variants we have currently
+			for _, tr := range trs {
+				// we didn't find an exact variant match
+				// next best guess is the first variant list that contains all of our known variants
+				if stringSubSlicesEqual(variants, tr.Variants) && tr.SuiteName == suite {
+					return &tr, nil
+				}
```

Comment:
- How is trs sorted? I think it's not deterministic which set of variants you'd pick, meaning different analyses could end up doing different comparisons.

- This seems to require we have at a test result that has *all* of the variants from the job run, but I think what we want is really a fuzzy best match kind of thing that works both ways.  If we get a job run that's "amd64, ha, fips" but we didn't have "fips" results, we'd be ok taking "amd64, ha" from the test results.

Reply:
Yes, this is non deterministic but it is also meant to be a last resort / better than 'unknown'.  I added it before we pulled in the job variant data so we may not even want it.

Reply:
I did want the early feedback on this so thanks.  I think the original case was we were missing the platform for the agnostic case.  So I added this so that we could match a test run (anyone really) that had at least the same set of variants that the incoming test had.  Ultimately with the addition of the job variant data coming in we probably don't need this but we can discuss other use cases and how we want this to behave overall.

Reply:
If you are saying I should add a third pass that flipped the superset check I can definitely do that.  I can also look to collect all the superset matches and the add some logic to which one gets returned maybe the fewest differences or a sorted list of the variant values.